File: src/game/level/world2/World2Level2.java
package game.level.world2;

import game.entity.enemies.npcs.Eagle;
import game.entity.enemies.npcs.TankEnemy;
import game.entity.models.Enemy;
import game.level.Level;

public class World2Level2 extends World2Level {
    @Override
    public int getLevelId() {
        return 2;
    }

    @Override
    public int startEnemiesCount() {
        return 8;
    }

    @Override
    public Class<? extends Enemy>[] availableEnemies() {
        return new Class[]{
                TankEnemy.class,
                Eagle.class,
        };
    }

    @Override
    public Class<? extends Level> getNextLevel() {
        return World2Level3.class;
    }
}

File: src/game/level/world2/World2Level2.kt
package game.level.world2

import game.entity.enemies.npcs.Eagle
import game.entity.enemies.npcs.TankEnemy
import game.entity.models.Enemy
import game.level.Level

class World2Level2 : World2Level() {
    override val levelId: Int
        get() {
            return 2
        }

    override val startEnemiesCount: Int
        get() {
            return 8
        }

    override val availableEnemies: Array<Class<out Enemy>>
        get() {
            return arrayOf(
                    TankEnemy::class.java,
                    Eagle::class.java)
        }

    override val nextLevel: Class<out Level?>
        get() {
            return World2Level3::class.java
        }
}